auth-service/src/main/kotlin/uk/gov/communities/delta/auth/services/PasswordTokenService.kt

```diff
@@ -78,24 +78,24 @@ abstract class PasswordTokenService(private val tableName: String, private val d
 
     suspend fun validateToken(token: String, userCN: String): TokenResult {
         return withContext(Dispatchers.IO) {
-            readToken(token, userCN, false)
+            readToken(token, userCN)
         }
     }
 
-    suspend fun consumeToken(token: String, userCN: String): TokenResult {
+    suspend fun consumeTokenIfValid(token: String, userCN: String): TokenResult {
         return withContext(Dispatchers.IO) {
-            readToken(token, userCN, true)
+            val tokenResult = readToken(token, userCN)
+            if (tokenResult is ValidToken) {
+                deleteToken(token, userCN)
+            }
+            tokenResult
```

Comment:
This could let two concurrent requests both consume the token, which doesn't really matter too much, but we should try and prevent it. `deleteToken` should check `executeUpdate` returns 1 to indicate a row was deleted.

Ideally the read would be a `SELECT FOR UPDATE` if we're consuming it to lock the row, and then the delete happens in the same transaction (so in one `useConnection` block), but that's a bit more faff.